### ISSUE TITLE
[$2000] iOS mWeb - Scroll behavior on some pages is inconsistent and broken

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#13599]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
Scenarios: 
1.Login with any account that has many reports.
2.From LHN, try to scroll to bottom with the first touch at the fab button, after that try to scroll normally at other position.
3.Notice you can't scroll right after that, you need to scroll 2,3 times to make it scroll as normal.

1.Select any workspace, choose Connect Bank Account => Connect Manually, press back button to Connect Bank Account, continue to press back.
2.Notice you can't scroll right after that, you need to scroll 2,3 times to make it scroll as normal.

1.Go to Settings => About page => Scroll to bottom, change the tab then go back.
2.Notice that you can’t scroll to bottom after that

## Expected:
Can scroll to the bottom as normal

## Actual:
Can not scroll to the bottom as normal

## Workaround:
unknown

## Platform:

Where is this issue occurring?

- Mobile Web Safari

**Version Number:** 1.2.39-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**Email or phone of affected tester (no customers):** 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/207710442-d7cae06b-8d10-431b-813b-b919b2331252.MP4

https://user-images.githubusercontent.com/43996225/207710473-30935a84-8929-4452-9b34-c6afcd813d9b.MP4

https://user-images.githubusercontent.com/43996225/208127073-6e5a262d-9995-41ac-a3d0-90bc455e4df3.MP4

https://user-images.githubusercontent.com/43996225/208127089-ea3f48f6-87eb-4b86-9a95-b523a3c16bc7.MP4

**Expensify/Expensify Issue URL:** 
**Issue reported by:** @hungvu193 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1671032426349589

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~012729407713b5d561</li>
        <li>Upwork Job ID: 1606017543317778432</li>
        <li>Last Price Increase: 2022-12-29</li>
    </ul>
</details>

**Comment by davidcardoza** [comment]
I was able to reproduce, sending out to external
https://user-images.githubusercontent.com/6643649/209216672-77a91412-be9d-4469-8d3d-30fa2d5b0ed7.MP4

**Comment by dhanashree-sawant** [comment]
**Proposal**

**Cause**:  IOS has feature of bounce on scroll whenever we reach extreme ends while scrolling. Whenever we scroll to bottom or top and content tries to bounce, on some occasion, due to the user trying to scroll beyond the content, it locks the parent View with it which changes touch focus to parent view and thus stopping the scroll.

**Solution**: add attribute bounces={false} in FlatList and ScrollView for that page which will stop the bounce on scroll feature and thus solve the issue. I will mention the code to change below for 'About' and 'Reports' page which are currently facing the issue. In future too, if such issue occurs again on any other page, only thing we will need to add is the bounces attritube to the component

```
src/pages/settings/AboutPage/AboutPage.js
@@ -54,26 +54,27 @@
 <ScrollView
                contentContainerStyle={[
                    styles.flexGrow1,
                    styles.flexColumn,
                    styles.justifyContentBetween,
                ]}
 +             bounces={false}
            >
```

```
src/components/InvertedFlatList/BaseInvertedFlatList.js
@@ -121,26 +121,27 @@
            <FlatList
                // eslint-disable-next-line react/jsx-props-no-spreading
                {...this.props}
                ref={this.props.innerRef}
                renderItem={this.renderItem}
+              bounces={false}
```

**Comment by Santhosh-Sellavel** [comment]
I'm not able to reproduce this issue @kavimuru please check on the latest staging again!

https://user-images.githubusercontent.com/85645967/209375794-c1b06d4f-621c-463b-9654-db673edbd5ef.MP4

**Comment by Santhosh-Sellavel** [comment]
@dhanashree-sawant 

Thanks for the proposal, but we need the bounce effect!

cc: @pecanoro

**Comment by Santhosh-Sellavel** [comment]
Waiting for proposals!

**Comment by pecanoro** [comment]
I can't reproduce anymore, but also I am using an emulator so I am not sure if it's different with a real device. @davidcardoza can you try again and see if you can reproduce it?

**Comment by hungvu193** [comment]
I'm still able to reproduce on latest versionhttps://user-images.githubusercontent.com/16502320/209892453-a88379c8-a117-4095-91ca-baf377e7a4d4.MOV

**Comment by davidcardoza** [comment]
Yeah, i can still reproduce the LHN freezing. Here's a new screen recording- 

https://user-images.githubusercontent.com/6643649/209904359-b433f5e7-8051-4a9a-8ec8-17024bbfeb44.MP4

**Comment by shonsirsha** [comment]
## Proposal

Hey there 👋 I think this issue can be solved in various ways, however I'd propose the minimal changes below to fix this issue of  **LHN "body lock" / "scroll freeze"**

(Brief explanation below this code)

File: `web/index.html`
```diff
<style>
      html,
      body,
      #root,
      #root > div,
      #root > div > div {
          height: 100% !important;
      }

+    body {
+         overflow:  hidden;         
+    }
....
```

The reason for such error (body locking) is because for some reason, on certain instances of iOS's Safari, the focus gets changed to document's `body` instead of the element we want to scroll. Therefore, when we're scrolling, it sometimes scrolls the actual `body` itself and causes the "body lock" / "scroll freeze" bug that we're trying to fix here.

Therefore, the simplest solution I can propose is the above, which is to "disable" the scrolling of the body by using `overflow: hidden` on the `body` element. The reason I chose to solve it this way is because it introduces the least amount of code and can be easily reverted, should it be necessary for any purpose.

[Preview] Before (scroll motions: up then down): 

https://user-images.githubusercontent.com/28676406/210034441-39f3c2dd-720e-4780-b92c-3d0394074f0a.MP4



[Preview] After (scroll motions: up then down):

https://user-images.githubusercontent.com/28676406/210034445-52c8c1eb-8e11-473a-852c-d6e2df524d89.MP4

Tested on Safari & Chrome (MacOS & iOS). 
_Note: there's a similar bug on iOS chrome and this fixed it too._

I hope this helps. Thanks! 🙂

**Comment by syedsaroshfarrukhdot** [comment]
It seems root cause is similar to this issue reported here on mWeb/Safari it was closed due to recognized as a safari bug and any solution would be a workaround to solve it https://github.com/Expensify/App/issues/13492. I believe we should do similar here.

**Comment by shonsirsha** [comment]
This bug is also similar to this: [White space appears under compose box](https://github.com/Expensify/App/issues/13219)

Worth noting that my solution above fixes the problem described in this issue (LHN). I can't repro the problem described in this issue in screens other than LHN.

**Comment by Santhosh-Sellavel** [comment]
Unable to test this one due to IP restriction, will check again tomorrow!

**Comment by Santhosh-Sellavel** [comment]
@Expensify/design 

Can someone check if are we fine with [this](https://github.com/Expensify/App/issues/13599#issuecomment-1367719421) new behavior.
@pecanoro All the solution looks fine to me, as it addresses the [issue](https://github.com/Expensify/App/issues/13599#issuecomment-1367068970) shown here, I don't think it breaks anything please let me know your thoughts, thanks!

**Comment by shonsirsha** [comment]
Thanks for the followups! 🙂 

@shawnborton , Ive been testing it the past 2 days and haven't found any issue that it caused - although it is hard to confirm that it won't cause any regression as there's no exact place on where it should be tested... 

Regarding your concern about blocking any scrolling though (which would be the regression), I think we are safe in that aspect because we never (nor we should) scroll directly from the html's `body`, but there's always a component that wraps the content in a page - mostly `View` (from `react-native`) which is where users scroll. We now also learn that we should never scroll the body, otherwise the exact bug we're trying to fix is gonna appear.

Anyhow, the proposal actually fixes the same issue that aren't even mentioned here which appears in different places, for example when clicking "new chat" on LHN, and in the "New chat". So most likely this proposal fixes this "body locking" scrolling bug everywhere, not just LHN.

Lastly, by any chance do you have any reference as to what needs to be tested further from your side? As, again, for now scrolling works fine everywhere. Thanks! 😄

**Comment by shawnborton** [comment]
Cool cool that sounds good, mostly just want to make sure all scrollable views/lists in the app will still scroll as expected!

**Comment by pecanoro** [comment]
Thank you both for your input! I think it's a great solution, but yeah, we will have to be careful about testing so we don't break any other parts of the UI! @shonsirsha assigning you to the job!

**Comment by shonsirsha** [comment]
> Cool cool that sounds good, mostly just want to make sure all scrollable views/lists in the app will still scroll as expected!

> I  think it's a great solution, but yeah, we will have to be careful about testing so we don't break any other parts of the UI

Totally agree! Will keep testing stuff! 😄 

Applied on Upwork, will create a PR today or latest tomorrow. Thanks!

**Comment by shonsirsha** [comment]
PR ready for review 🙂 
cc @Santhosh-Sellavel @shawnborton

**Comment by davidcardoza** [comment]
@pecanoro What is the next step on this one?

**Comment by pecanoro** [comment]
@davidcardoza Oh dammit, I think the issue wasn't linked properly on the PR and it didn't update the issue with the payment date. You can follow the steps [here](https://stackoverflowteams.com/c/expensify/questions/8582/8583#8583) from step 6. You will need to pay the people involved since it's been more than a week since we deployed the changes. If you have any doubts about how to proceed, I recommend asking on the #bug-zero channel.

**Comment by shonsirsha** [comment]
@0xmiroslav thanks for reporting. Although I did quite a thorough testing for this one, I missed it as I didn't know that the regression is even a functionality in Android web. Sorry about this.

Anyhow, I don't find another way to solve this in iOS other than disabling the scrolling of the document's body. I'm now thinking of somehthing like this which should fix the regression yet still fixes the issue:


File: `index.html`
```html
<script>
function disableBodyScrollIOS(){
      if (isMobileSafari()){
            document.body.style.overflow = 'hidden';
      }
}
</script>

<body onload="disableBodyScrollIOS();">
etc...
```

`isMobileSafari` is a function that we have in the codebase that we can copy over to index.html.

Open for suggestions.
cc @pecanoro @Santhosh-Sellavel 

Also would anyone lmk how to proceed if a PR causes regression? Thx.

**Comment by Santhosh-Sellavel** [comment]
> [PR](https://github.com/Expensify/App/pull/14005) which fixes this issue and causes serious regression in android chrome. Pull-to-refresh not working. Issue reported here: https://expensify.slack.com/archives/C049HHMV9SM/p1674040409891389

Good catch @0xmiroslav. But I think it is not a serious regression as it does not affect anything within our app. I will wait for @pecanoro to decide on this one.

Whether we fix it or not I think this is something that is a bit out of scope to find out in the PR review testing, thanks!

**Comment by Santhosh-Sellavel** [comment]
Thanks, everyone, @davidcardoza this is due for payment!

**Comment by davidcardoza** [comment]
Confirming payments with @pecanoro  before sending out.

**Comment by davidcardoza** [comment]
- Payment sent to @shonsirsha 
- @hungvu193 I hired you on the job, please accept to receive $250 payment
- @Santhosh-Sellavel please apply for the job so I can process payment - https://www.upwork.com/jobs/~012729407713b5d561

**Comment by Santhosh-Sellavel** [comment]
@davidcardoza Applied now!

**Comment by Santhosh-Sellavel** [comment]
@davidcardoza bump!

**Comment by mallenexpensify** [comment]
Hired @Santhosh-Sellavel can you please accept the job and reply here once you have? 
https://www.upwork.com/jobs/~012729407713b5d561

**Comment by Santhosh-Sellavel** [comment]
Thanks, @mallenexpensify done!

**Comment by mallenexpensify** [comment]
Paid @Santhosh-Sellavel 

@davidcardoza and @Santhosh-Sellavel , can you fill out the below?  The checklist didn't auto-populate

BugZero Checklist: The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] The PR that introduced the bug has been identified. Link to the PR:
- [ ] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] A [regression test](https://expensify.testrail.io/index.php?/dashboard) has been added or updated so that the same bug will not reach production again. Link to the GH issue for creating the test here:

**Comment by Santhosh-Sellavel** [comment]
@pecanoro I believe this is not a regression, it existed right from the start!

**Comment by davidcardoza** [comment]
Thanks for taking over while i was away @mallenexpensify I filled out the two first bullet points.

**Comment by pecanoro** [comment]
@davidcardoza The bug-zero assignee usually takes care of the last bullet point, it would be great if you can take care of that last item, thank you!

**Comment by MelvinBot** [comment]
@pecanoro, @davidcardoza, @shonsirsha, @Santhosh-Sellavel Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by davidcardoza** [comment]
This doesn't require a regression test  it doesn’t make sense to add a step to TestRail test case.

**Comment by mallenexpensify** [comment]
I added it to the testing guidelines sheet  https://docs.google.com/document/d/1UaWlcieiwLnsKfeKw7iwLF7vgYRmdw_-K8OcY5aPBRU/edit#heading=h.zm638pdvzx2
